### ISSUE TITLE
[Improvement] Expose locale information such as Text Direction from Paraglide output

### ISSUE THREAD

**Issue by LorisSigrist** [inlang/monorepo#1768]
## Problem 

A common i18n requirement is to set the text-direction according to the locale. The `Intl.Locale` API aims to solve that, but it's poorly supported & inconsistently implemented across browser. 

This leaves users with two choices:
- A handcoded map of which locales have which conventions (text direction, calendar, first-day-of-week etc)
- Use a Polyfill, which can be quite large

## Proposal
As a compiler, paraglide is uniquely positioned, since it can make certain assumptions about it's runtime. We can be fairly confident that paraglide is running in some V8 powered runtime. We also aren't bundle-size limited, so loading a polyfill isn't a problem.  

We could determine information such as `textDirection` and Calendar Info at build time, and make it available.

This could be done through the addition of a `locales.js` file and `locales/{locale}.js` files. Here is an example output:
```js
// locale.js
export * as en from "./locales/en.js"
export * as ar from "./locales/ar.js"
```

```js
// locales/en.js
export const textDirection = "ltr"
export const ...


// locales/ar.js
export const textDirection = "rtl"
export const ...
```

This could then be used like
```js
import { languageTag } from "./paraglide/runtime.js"
import * as locales from "./paraglide/locales.js"

const dir = locales[languageTag()].textDirection
```
This would still be fully treeshakeable & typesafe.

## Alternatives
We could just leave this up to the users.

If you have a better Idea to cover this use-case, I'm all ears!

**Comment by LorisSigrist** [comment]
@samuelstroschein I would love to have your thoughts on this

**Comment by samuelstroschein** [comment]
@LorisSigrist this is a good example of a crucial ping to me because it touches on API decisions. 

Initial reaction: 

- this seems like a nice to have but is not crucial 
- we shouldn't add nice to have stuff in paraglide that will bloat the API of paraglide
- the proposal doesn't fix the issue of inconsistency across browsers

If the following is the problem: 

> The Intl.Locale API aims to solve that, but it's poorly supported & inconsistently implemented across browser.
> This leaves users with two choices:
> - A handcoded map of which locales have which conventions (text direction, calendar, first-day-of-week etc)
> - Use a Polyfill, which can be quite large

Your proposal doesn't solve it.

You assume V8 only (now you get the inconsistencies across browsers) and that a custom implementation instead of Intl.Locale is better. In other words, you are proposing a polyfill (the downside you described). 

Conclusion: 

I vote to go all in on`Intl` APIs which will be greatly extended with the upcoming [message format 2.0](https://github.com/unicode-org/message-format-wg/tree/main). Users have to polyfill/we can polyfill in the bundler if required. On the flipside, we don;t bloat paraglide with nice to have stuff that eventually falls apart due to

- API complexity (why can't I use standard JS?)
- communication complexity (why is the bundle size of paraglide larger?)
- maintenance effort unjustified (it seems like our custom solution will run into the same problems as the Intl API)

**Comment by LorisSigrist** [comment]
I agree that there are quite a few drawbacks putting this in the core. However I don't quite agree with everything you said.

> the proposal doesn't fix the issue of inconsistency across browsers

The values would be generated at build-time and then shipped to the browser. This _would_ eliminated inconsistencies. (I probably could have made this clearer)

> You assume V8 only (now you get the inconsistencies across browsers) and that a custom implementation instead of Intl.Locale is better. In other words, you are proposing a polyfill (the downside you described).

Yes, what I am proposing would likely require using a polyfill. The difference is it would be used at built time and not shipped to the client. This way we could guarantee consistent output regardless of browser or runtime, _without_ increasing bundle size (treeshaking). 

## Intl isn't there yet
I would prefer using the `Intl` Api. Using what's there pretty much always beats doing it yourself. 
But let me walk you through an example on how to do something simple like get the text-direction from `Intl`:

Firefox: You can't
Chrome: `(new Intl.Locale(languageTag())).textInfo.direction` <- not spec compliant
Safari: `(new Intl.Locale(languageTag())).getTextInfo().direction`

These kinds of inconsistencies exist all accross the `Intl` API. I'm certain they will get better but in the here and now it's really frustrating to work with. A polyfill is basically mandatory.

## Alternative
When building the Adapters, we will need to use this functionality anyway to properly manage the `<html dir>` attribute. We could just include any needed polyfills in the adapters.

**Comment by samuelstroschein** [comment]
Maybe turn this into a feature request? 

If users really complain hard about this, we could think of adding something. Don't fall into the trap of implementing every feedback you get in early stages. The risk is high that you will add features that don't work well on a larger scale but can't be removed anymore. We don't want this https://github.com/solidjs/solid/discussions/845#discussioncomment-2146959 to happen to Paraglide.

> The difference is it would be used at built time and not shipped to the client. This way we could guarantee consistent output regardless of browser or runtime, without increasing bundle size (treeshaking).

Adding code that is not built-in to a browser is increasing the bundle size, regardless if tree-shakable or not ;)
